contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1301 @@
+        require(address(mPool.tokenB()) == OETHp, "OETHp not TokanB");
+
+        // prevent implementation contract to be governed
+        _setGovernor(address(0));
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     */
+    function initialize() external onlyGovernor initializer {
+        // Read reward
+        address[] memory _rewardTokens = new address[](1);
+        _rewardTokens[0] = poolDistributor.rewardToken();
+
+        require(_rewardTokens[0] != address(0), "No reward token configured");
+
+        InitializableAbstractStrategy._initialize(
+            _rewardTokens,
+            new address[](0),
+            new address[](0)
+        );
+    }
+
+    /***************************************
+                  Configuration 
+    ****************************************/
+
+    /**
+     * @notice Set allowed pool weth share interval. After the rebalance happens
+     * the share of WETH token in the ticker needs to be withing the specifications
+     * of the interval.
+     *
+     * @param _allowedWethShareStart Start of WETH share interval expressed as 18 decimal amount
+     * @param _allowedWethShareEnd End of WETH share interval expressed as 18 decimal amount
+     */
+    function setAllowedPoolWethShareInterval(
+        uint256 _allowedWethShareStart,
+        uint256 _allowedWethShareEnd
+    ) external onlyGovernor {
+        require(
+            _allowedWethShareStart < _allowedWethShareEnd,
+            "Invalid interval"
+        );
+        // can not go below 1% weth share
+        require(_allowedWethShareStart > 0.01 ether, "Invalid interval start");
+        // can not go above 95% weth share
+        require(_allowedWethShareEnd < 0.95 ether, "Invalid interval end");
+
+        allowedWethShareStart = _allowedWethShareStart;
+        allowedWethShareEnd = _allowedWethShareEnd;
+        emit PoolWethShareIntervalUpdated(
+            allowedWethShareStart,
+            allowedWethShareEnd
+        );
+    }
+
+    /***************************************
+               Strategy overrides 
+    ****************************************/
+
+    /**
+     * @notice Deposits funds to the strategy which deposits them to the
+     * underlying Rooster pool if the pool price is within the expected interval.
+     * @param _asset   Address for the asset
+     * @param _amount  Units of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @notice Deposits all the funds to the strategy which deposits them to the
+     * underlying Rooster pool if the pool price is within the expected interval.
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance > 1e12) {
```

Comment:
nit: Same here for magic number (`1e12`).

Reply:
great point fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/9ad942f2df69f6d2ccd66e15305a51c319ec3237